src/vs/workbench/contrib/terminalContrib/clipboard/browser/terminalClipboard.ts

```diff
@@ -9,14 +9,84 @@ import { IDialogService } from '../../../../../platform/dialogs/common/dialogs.j
 import { ServicesAccessor } from '../../../../../platform/instantiation/common/instantiation.js';
 import { TerminalSettingId } from '../../../../../platform/terminal/common/terminal.js';
 
-export async function shouldPasteTerminalText(accessor: ServicesAccessor, text: string, bracketedPasteMode: boolean | undefined): Promise<boolean | { modifiedText: string }> {
+class SmartPasteUtils {
+	/**
+	 *
+	 * @param string
+	 * @returns true if the string is a path
+	 */
+	static isPath(string: string): boolean {
+		// Regex to detect common path formats
+		const windowsPathPattern = /^[a-zA-Z]:(\\|\/)/;  // Windows absolute path
+		const uncPathPattern = /^\\\\/;             // Windows UNC path
+		const unixPathPattern = /^\/|(\w+\/)/;      // Unix/Linux/macOS paths
+
+		return windowsPathPattern.test(string) ||
+			uncPathPattern.test(string) ||
+			unixPathPattern.test(string);
+	}
+
+	/**
+	 *
+	 * @param string
+	 * @param shellType
+	 * @returns func to handle smart paste, escape chars or wrap inside double quotes
+	 */
+	static handleSmartPaste(string: string, shellType: string): string {
+		if (!SmartPasteUtils.isPath(string)) {
+			return string;  // Return the string as is if it's not detected as a path
+		}
+
+		switch (shellType) {
+			case 'gitbash':
+			case 'cmd':
+				{
+					// Escape backslashes and wrap in double quotes if necessary
+					const escapedPath = string.replace(/\\/g, '\\\\');
+					if (string.includes(' ')) {
+						return `"${escapedPath}"`;
+					}
+					return escapedPath;
+				}
+
+			case 'bash':  // Linux/macOS Bash
+				// Wrap in quotes if spaces or special characters exist
+				if (string.includes(' ')) {
+					return `"${string}"`;
+				}
+				return string;
+
+			case 'pwsh':
+				// Simply wrap in quotes if spaces are present
+				if (string.includes(' ')) {
+					return `"${string}"`;
+				}
```

Comment:
You also need to ensure that the quote you're using is escaped properly within `string`